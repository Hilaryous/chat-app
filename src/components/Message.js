// @flow
import React from 'react'
import styled from 'react-emotion'
import type MessageType from '../types/Message'

type Props = {
  message: MessageType,
}

const Message = ({ message }: Props) => (
  <Container>
    <Author>
      {message.userEmail}
    </Author>
    {message.content}
  </Container>
)

export default Message

// --------------------------

const Container = styled('div')({
  fontFamily: 'sans-serif',
  padding: '16px',
})

const Author = styled('div')({
  fontWeight: 'bold',
})
