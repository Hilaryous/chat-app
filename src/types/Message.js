// @flow
export type MessageType = {
  id: string,
  chatId: string,
  userId: string,
  content: string,
  userEmail?: string,
}
