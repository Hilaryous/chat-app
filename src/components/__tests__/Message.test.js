import React from 'react'
import renderer from 'react-test-renderer'
import Message from '../Message'

describe('Message', () => {
  it('shows the correct snapshot tree for the Message component', () => {
    const tree = renderer.create(
      <Message
        message={{ userEmail: 'foo@test.com', content: 'test' }}
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
