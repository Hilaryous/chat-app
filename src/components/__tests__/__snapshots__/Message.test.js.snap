// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message shows the correct snapshot tree for the Message component 1`] = `
<div
  className="css-17cr7tx"
>
  <div
    className="css-in3yi3"
  >
    foo@test.com
  </div>
  test
</div>
`;
